### PR TITLE
Change order of the floating operation to match prediction in native xgboost version 1.3.0 and newer

### PR DESCRIPTION
**Pull request by valenad1**
As its described [here](https://github.com/dmlc/xgboost/issues/6350), the floating point order of the operation has been changed in version 1.3.0. I have adjusted the predictor according to this changes (base_score is incremented by tree prediction instead of tree prediction incremented by base_score at the end)

### CODE REVIEW

File: xgboost-predictor/src/main/java/biz/k11i/xgboost/gbm/GBLinear.java [hunk 1]
```diff
@@ -26,7 +26,7 @@ public void loadModel(PredictorConfiguration config, ModelReader reader, boolean
     }
 
     @Override
-    public float[] predict(FVec feat, int ntree_limit) {
+    public float[] predict(FVec feat, int ntree_limit, float base_score) {
```
Comment (mn-mikke) [inline xgboost-predictor/src/main/java/biz/k11i/xgboost/gbm/GBLinear.java:29]: `base_score` is not applied in the logic. Or is it irrelevant for `GBLinear`?

File: xgboost-predictor/src/main/java/biz/k11i/xgboost/gbm/GBLinear.java [hunk 2]
```diff
@@ -35,7 +35,7 @@ public float[] predict(FVec feat, int ntree_limit) {
     }
 
     @Override
-    public float predictSingle(FVec feat, int ntree_limit) {
+    public float predictSingle(FVec feat, int ntree_limit, float base_score) {
```
Comment (mn-mikke) [inline xgboost-predictor/src/main/java/biz/k11i/xgboost/gbm/GBLinear.java:38]: ditto

File: xgboost-predictor/src/test/java/biz/k11i/xgboost/PredictorSmokeTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        /*
+         * obtained from xgboost version 1.3.0 and newer (Current upper bound is 1.6.1)
+         * Difference is due the different sequence of floating point addition
+         * see: https://github.com/dmlc/xgboost/issues/6350
+         * Previous result was 64.41069031f obtained from xgboost version 1.2.0
+         */
+        float expected = 64.41068268f;
+
+        Assert.assertEquals(1, prediction.length);
+        Assert.assertEquals(expected, prediction[0], 0);
```
Comment (mn-mikke) [inline xgboost-predictor/src/test/java/biz/k11i/xgboost/PredictorSmokeTest.java:72]: shouldProvideDifferentPrediction => where is assert that the numbers are different

Comment (valenad1) [reply xgboost-predictor/src/test/java/biz/k11i/xgboost/PredictorSmokeTest.java:72]: 64e5f466a301c719

Comment (valenad1) [reply xgboost-predictor/src/test/java/biz/k11i/xgboost/PredictorSmokeTest.java:72]: fixed

File: xgboost-predictor/src/main/java/biz/k11i/xgboost/gbm/GBLinear.java (hunk 1 above)
Comment (valenad1) [reply xgboost-predictor/src/main/java/biz/k11i/xgboost/gbm/GBLinear.java:29]: I am working on tests for Linear and Dart implementation, we don't cover it nor here nor in H2O-3.

Good point

**Comment by valenad1** [comment]
Hi @mn-mikke, all booster implementation should be fixed now

**Review by mn-mikke — APPROVED** [review]
LGTM, thank you!

**Pull request merged**